src/terraform_gui.cpp

```diff
@@ -149,6 +155,34 @@ void PlaceProc_DemolishArea(TileIndex tile)
 	VpStartPlaceSizing(tile, VPM_X_AND_Y, DDSP_DEMOLISH_AREA);
 }
 
+/**
+ * Show a dropdown list for the demolition tool.
+ * @param w Parent window for the list
+ */
+static void ToolbarDemolishClick(Window *w)
+{
+	DropDownList *list = new DropDownList();
+	DropDownListStringItem *item;
+
+	item = new DropDownListStringItem(STR_DEMOLISH_EVERYTHING, 0, false);
+	list->push_back(item);
+	item = new DropDownListStringItem(STR_DEMOLISH_TREES, 1, false);
+	list->push_back(item);
+	item = new DropDownListStringItem(STR_DEMOLISH_COMPANY_PROPERTIES, 2, false);
+	list->push_back(item);
+
+	ShowDropDownList(w, list, 0, WID_TT_DEMOLISH, 140, true, true);
+}
```

Comment:
* DropDownLists have changed, use `DropDownList list;` instead.
* Don't use a temporary item variable, just `list.emplace_back(new ...);`
* Use enum values instead of 0-1-2